### PR TITLE
SO-55-home-page-testimonial

### PR DESCRIPTION
**Pull request by hashemJaber**
Can someone code review?

### CODE REVIEW

File: src/app/(dashboard)/dashboard/page.tsx
Comment (jvniorrr) [inline src/app/(dashboard)/dashboard/page.tsx:1]: Old code should be one referenced here

File: src/app/page.tsx [hunk 1]
```diff
@@ -28,8 +30,44 @@ const cardTwoContent = (
 	</ul>
 	// {/* </div> */}
 );
+interface Testimonial {
```
Comment (jvniorrr) [inline src/app/page.tsx]: If possible move Testimonial Table interface into the file `@/types/database.interface.ts`

File: src/app/page.tsx [hunk 2]
```diff
@@ -28,8 +30,44 @@ const cardTwoContent = (
...
+	id: number;
+	created_at: string;
+	review_content: string;
+	user_id: number;
+	userName: string;
+	user_profile_link: string;
+  }
+  
+function returnNewCard(testimonial : Testimonial ){
```
Comment (jvniorrr) [inline src/app/page.tsx]: Move component to a component file under the @/components directory.

File: src/app/page.tsx [hunk 3]
```diff
@@ -56,8 +94,34 @@ const cardThreeContent = (
 );
 
 export default function Home() {
+	
+	const [thirdCard_tes,setThirdCard_tes]= useState(null);
+	
+	useEffect(()=>{
```
Comment (jvniorrr) [inline src/app/page.tsx]: Try and see if we can move the card fetching of data into its own component that way we can keep the highest component, in this case page.tsx as a server side component. So any logic using useState or useEffect, lets try and make those directly under components as client side components and import into the server side component.s  If questions on this please ask.

File: src/components/Login/LoginForm.tsx [hunk 4]
```diff
@@ -39,8 +39,6 @@ function LoginForm() {
 		}
 
 		// redirect to dashboard
-		// assure client components get refreshed
-		router.refresh();
 		router.push('/dashboard');
```
Comment (jvniorrr) [inline src/components/Login/LoginForm.tsx]: Use old code
```suggestion
		// assure client components get refreshed
		router.refresh();
		router.push('/dashboard');
```

File: src/components/Dropdown.tsx
Comment (jvniorrr) [inline src/components/Dropdown.tsx:1]: Use old code base here 
```
<Listbox.Options className="absolute max-h-60 overflow-auto bg-white rounded-md -left-4">
						{languages.map(language => (
							<Listbox.Option
								key={language.id}
								value={language}
								disabled={language.unavailable}
								className={({ active }) =>
									`${
										selectedLanguage.name ===
											language.name || active
											? 'font-bold '
											: ''
									} dropdown-item`
								}
							>
```

**Review by jvniorrr — COMMENTED** [review]
Good implementation, just move to appropriate pages. Lets try and keep the top level files as static / server side components to make use of NextJS fully & reduce client side rendering or performance hits. Besides that thank you sir! 
Fresh

**Comment by hashemJaber** [comment]
yes sir 
alt text

File: src/app/page.tsx (hunk 2 above)
Comment (hashemJaber) [reply src/app/page.tsx]: Done

File: src/app/page.tsx (hunk 3 above)
Comment (hashemJaber) [reply src/app/page.tsx]: Done and Done

File: src/components/HomePage/Testimonial.tsx
Comment (jvniorrr) [inline src/components/HomePage/Testimonial.tsx:1]: good work

File: src/components/HomePage/Testimonial.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+	const [error, setError] = useState(null);
+	const num_rand = 5;
+
+	useEffect(() => {
+		const fetchTest = async () => {
+			setIsLoading(true);
+			try {
+				if (!thirdCard_tes.length) {
+					const supabase = await createSupbaseClient(
+						process.env.NEXT_PUBLIC_SUPABASE_URL!,
```
Comment (jvniorrr) [inline src/components/HomePage/Testimonial.tsx]: remove uneccessary params <3

**Review by jvniorrr — CHANGES_REQUESTED** [review]

**Review by jvniorrr — APPROVED** [review]

**Comment by hashemJaber** [comment]
SO-55 Passed Code Review

**Pull request merged**